src/app/loginComponents/accounts/internal/accounts.component.ts

```diff
@@ -14,41 +16,37 @@ import { UsersService } from './../../../shared/swagger/api/users.service';
 })
 export class AccountsInternalComponent implements OnInit {
   user: User;
+  TokenSource = TokenSource;
   googleProfile: Profile;
   gitHubProfile: Profile;
   hasGitHubToken$: Observable<boolean>;
   hasGoogleToken$: Observable<boolean>;
-  public syncingWithGitHub = false;
-  public syncingWithGoogle = false;
+  show: false;
+  public syncing = false;
   constructor(private userService: UserService, private usersService: UsersService, private configuration: Configuration,
-    private authService: AuthService, private tokenService: TokenService) {
+    private authService: AuthService, private tokenService: TokenService, private matSnackBar: MatSnackBar) {
       this.hasGitHubToken$ = this.tokenService.hasGitHubToken$;
       this.hasGoogleToken$ = this.tokenService.hasGoogleToken$;
     }
 
-  syncGitHub() {
-    this.syncingWithGitHub = true;
-    this.usersService.updateLoggedInUserMetadata('github.com').subscribe((user: User) => {
+  /**
+   * Update user metadata for a service ('github.com' or 'google.com')
+   *
+   * @param {string} service  'github.com' or 'google.com'
+   * @memberof AccountsInternalComponent
+   */
+  sync(service: TokenSource) {
+    this.syncing = true;
+    this.usersService.updateLoggedInUserMetadata(service).subscribe((user: User) => {
```

Comment:
I can't comment on lines you didn't change, but since you're no longer hard-coding the token type here, do you want to change lines 60 and 65 to use`TokenSource.GOOGLE.toString()` and `TokenSource.GITHUB.toString()`?

Reply:
sure, might as well :+1: